diff --git a/data/sql/world/world_playerbots_rpg_races.sql b/data/sql/world/world_playerbots_rpg_races.sql
index 75e45dd5..d5d6ee96 100644
--- a/data/sql/world/world_playerbots_rpg_races.sql
+++ b/data/sql/world/world_playerbots_rpg_races.sql
@@ -39,17 +39,20 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 6734, 3, 10, 20),
 (NULL, 6734, 7, 10, 20),
 --
---       NIGHT ELVES
+--       NIGHT ELVES AND WORGEN
 --
 -- Innkeeper Keldamyr, Dolanaar
 (NULL, 6736, 4, 1, 10),
+(NULL, 6736, 12, 1, 10),
 -- Innkeeper Shaussiy, Auberdine
 (NULL, 6737, 4, 10, 20),
+(NULL, 6737, 12, 10, 20),
 --
 --       ALLIANCE CITIES
 --
 -- Innkeeper Saelienne, Darnassus
 (NULL, 6735, 4, 10, 80), -- elves
+(NULL, 6735, 12, 10, 80), -- worgen
 --
 -- Innkeeper Firebrew, Ironforge
 (NULL, 5111, 3, 10, 80), -- dwarves
@@ -57,6 +60,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 5111, 4, 10, 80), -- elves
 (NULL, 5111, 1, 10, 80), -- human
 (NULL, 5111, 11, 20, 80), -- draenei
+(NULL, 5111, 12, 10, 80), -- worgen
 --
 -- Innkeeper Allison, Stormwind
 (NULL, 6740, 1, 10, 80), -- human
@@ -64,6 +68,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 6740, 4, 10, 80), -- elves
 (NULL, 6740, 7, 10, 80), -- gnomes
 (NULL, 6740, 11, 20, 80), -- draenei
+(NULL, 6740, 12, 10, 80), -- worgen
 --
 -- Caregiver Breel Exodar
 (NULL, 16739, 11, 10, 80), -- draenei
@@ -71,6 +76,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 16739, 3, 60, 80), -- dwarves
 (NULL, 16739, 4, 60, 80), -- elves
 (NULL, 16739, 7, 60, 80), -- gnomes
+(NULL, 16739, 12, 60, 80), -- worgen
 --
 --       ALLIANCE CONTESTED LOCATIONS
 --
@@ -78,20 +84,25 @@ INSERT INTO `playerbots_rpg_races` VALUES
 --
 (NULL, 6738, 4, 15, 30), -- elves
 (NULL, 6738, 11, 20, 30), -- draenei
+(NULL, 6738, 12, 15, 30), -- worgen
 --
 -- Innkeeper Faralia, Stonetalon Peak
 --
 (NULL, 16458, 4, 15, 27), -- elves
+(NULL, 16458, 12, 15, 27), -- worgen
 --
 -- Innkeeper Lyshaerya, Desolase
 --
 (NULL, 11103, 4, 30, 40), -- elves
+(NULL, 11103, 12, 30, 40), -- worgen
 --
 -- Innkeeper Shyria, Feathermoon, Feralas
 (NULL, 7736, 4, 40, 50), -- elves
+(NULL, 7736, 12, 40, 50), -- worgen
 --
 -- Falfindel Waywarder, Feralas elf camp
 (NULL, 4048, 4, 40, 50), -- elves
+(NULL, 4048, 12, 40, 50), -- worgen
 --
 -- Innkeeper Helbrek, Wetlands
 --
@@ -104,6 +115,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 6790, 3, 18, 30), -- dwarves
 (NULL, 6790, 7, 18, 30), -- gnomes
 (NULL, 6790, 11, 20, 30), -- draenei
+(NULL, 6790, 12, 20, 30), -- worgen
 --
 -- Innkeeper Brianna, Redridge Mountains
 --
@@ -114,6 +126,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 2352, 3, 20, 30), -- dwarves
 (NULL, 2352, 7, 20, 30), -- gnomes
 (NULL, 2352, 11, 20, 30), -- draenei
+(NULL, 2352, 12, 20, 30), -- worgen
 --
 -- Captain Nials, Refuge Pointe, Arathi
 (NULL, 2700, 1, 30, 40), -- human
@@ -121,6 +134,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 2700, 7, 30, 40), -- gnomes
 (NULL, 2700, 4, 30, 40), -- elves
 (NULL, 2700, 11, 30, 40), -- draenei
+(NULL, 2700, 12, 30, 40), -- elves
 --
 -- Lt. Doren, Stranglethorn Vale
 (NULL, 469, 1, 30, 45), -- human
@@ -128,12 +142,14 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 469, 4, 30, 45), -- elves
 (NULL, 469, 7, 30, 45), -- gnomes
 (NULL, 469, 11, 30, 45), -- draenei
+(NULL, 469, 12, 30, 45), -- worgen
 --
 -- Innkeeper Janene, Theramore
 (NULL, 6272, 1, 35, 45), -- human
 (NULL, 6272, 3, 35, 45), -- dwarves
 (NULL, 6272, 7, 35, 45), -- gnomes
 (NULL, 6272, 11, 35, 45), -- draenei
+(NULL, 6272, 12, 35, 45), -- worgen
 --
 -- Innkeeper Prospector Ryedol, Badlands Q-giver
 (NULL, 2910, 3, 35, 45), -- dwarves
@@ -148,6 +164,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 11548, 4, 45, 55), -- elves
 (NULL, 11548, 1, 45, 55), -- human
 (NULL, 11548, 11, 45, 55), -- draenei
+(NULL, 11548, 12, 45, 55), -- worgen
 --
 -- Thadius Grimshade, Nethergarde Keep, Blasted Lands
 (NULL, 8022, 1, 45, 55), -- human
@@ -155,6 +172,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 8022, 4, 45, 55), -- elves
 (NULL, 8022, 7, 45, 55), -- gnomes
 (NULL, 8022, 11, 45, 55), -- draenei
+(NULL, 8022, 12, 45, 55), -- worgen
 --
 -- Gothine the Hooded, Felwood Alliance camp
 (NULL, 9465, 4, 48, 55), -- elves
@@ -162,6 +180,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 9465, 3, 48, 55), -- dwarves
 (NULL, 9465, 7, 48, 55), -- gnomes
 (NULL, 9465, 11, 48, 55), -- draenei
+(NULL, 9465, 12, 48, 55), -- worgen
 --
 -- Muigin, Alliance Q-giver, Un'Goro
 (NULL, 9119, 1, 48, 55), -- human
@@ -169,6 +188,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 9119, 4, 48, 55), -- elves
 (NULL, 9119, 7, 48, 55), -- gnomes
 (NULL, 9119, 11, 48, 55), -- draenei
+(NULL, 9119, 12, 48, 55), -- draenei
 --
 -- Alchemist Arbington, West Plaguelands, Human
 (NULL, 11056, 1, 51, 58), -- human
@@ -176,6 +196,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 11056, 4, 51, 58), -- elves
 (NULL, 11056, 7, 51, 58), -- gnomes
 (NULL, 11056, 11, 51, 58), -- draenei
+(NULL, 11056, 12, 51, 58), -- worgen
 --
 -- Borgus Stourarm, Alliance Taxi, Burning Steppes
 (NULL, 2299, 1, 50, 60), -- human
@@ -183,6 +204,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 2299, 4, 50, 60), -- elves
 (NULL, 2299, 7, 50, 60), -- gnomes
 (NULL, 2299, 11, 50, 60), -- draenei
+(NULL, 2299, 12, 50, 60), -- worgen
 --
 -- Marshal Bluewall, Alliance camp, Silithus
 (NULL, 17080, 1, 55, 60), -- human
@@ -190,6 +212,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 17080, 4, 55, 60), -- elves
 (NULL, 17080, 7, 55, 60), -- gnomes
 (NULL, 17080, 11, 55, 60), -- draenei
+(NULL, 17080, 12, 55, 60), -- worgen
 --
 --           OUTLAND
 --
@@ -199,6 +222,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 19229, 4, 58, 59), -- elves
 (NULL, 19229, 7, 58, 59), -- gnomes
 (NULL, 19229, 11, 58, 59), -- draenei
+(NULL, 19229, 12, 58, 59), -- worgen
 --
 -- Sid Limbardi, Honor Hold, Hellfire
 (NULL, 16826, 1, 58, 63), -- human
@@ -206,6 +230,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 16826, 4, 58, 63), -- elves
 (NULL, 16826, 7, 58, 63), -- gnomes
 (NULL, 16826, 11, 58, 63), -- draenei
+(NULL, 16826, 12, 58, 63), -- worgen
 --
 -- Caregiver Ophera Windfury, Draenei, Hellfire
 (NULL, 18906, 1, 60, 63), -- human
@@ -213,6 +238,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 18906, 4, 60, 63), -- elves
 (NULL, 18906, 7, 60, 63), -- gnomes
 (NULL, 18906, 11, 60, 63), -- draenei
+(NULL, 18906, 12, 60, 63), -- worgen
 --
 -- Caregiver Abidaar, Telredor, Zangarmarsh
 (NULL, 18251, 1, 60, 63), -- human
@@ -220,6 +246,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 18251, 4, 60, 63), -- elves
 (NULL, 18251, 7, 60, 63), -- gnomes
 (NULL, 18251, 11, 60, 63), -- draenei
+(NULL, 18251, 12, 60, 63), -- worgen
 --
 -- Caregiver Kerp, Orebor, Zangarmarsh
 (NULL, 18908, 1, 61, 64), -- human
@@ -227,6 +254,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 18908, 4, 61, 64), -- elves
 (NULL, 18908, 7, 61, 64), -- gnomes
 (NULL, 18908, 11, 61, 64), -- draenei
+(NULL, 18908, 12, 61, 64), -- worgen
 --
 -- Innkeeper Biribi, Terrokar
 (NULL, 19296, 1, 62, 65), -- human
@@ -234,6 +262,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 19296, 4, 62, 65), -- elves
 (NULL, 19296, 7, 62, 65), -- gnomes
 (NULL, 19296, 11, 62, 65), -- draenei
+(NULL, 19296, 12, 62, 65), -- worgen
 --
 -- Caregiver Isel, Telaar, Nagrand
 (NULL, 18914, 1, 64, 67), -- human
@@ -241,6 +270,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 18914, 4, 64, 67), -- elves
 (NULL, 18914, 7, 64, 67), -- gnomes
 (NULL, 18914, 11, 64, 67), -- draenei
+(NULL, 18914, 12, 64, 67), -- worgen
 --
 -- Innkeeper Shaunessy, Sylvanaar, Blade's Edge
 (NULL, 19495, 1, 65, 68), -- human
@@ -248,6 +278,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 19495, 4, 65, 68), -- elves
 (NULL, 19495, 7, 65, 68), -- gnomes
 (NULL, 19495, 11, 65, 68), -- draenei
+(NULL, 19495, 12, 65, 68), -- worgen
 --
 -- Innkeeper Fizir Doc Clocktock, Blade's Edge
 (NULL, 21110, 1, 65, 68), -- human
@@ -255,6 +286,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 21110, 4, 65, 68), -- elves
 (NULL, 21110, 7, 65, 68), -- gnomes
 (NULL, 21110, 11, 65, 68), -- draenei
+(NULL, 21110, 12, 65, 68), -- worgen
 --
 -- Innkeeper Dreg Cloudsweeper, Shadowmoon
 (NULL, 19352, 1, 67, 70), -- human
@@ -262,6 +294,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 19352, 4, 67, 70), -- elves
 (NULL, 19352, 7, 67, 70), -- gnomes
 (NULL, 19352, 11, 67, 70), -- draenei
+(NULL, 19352, 12, 67, 70), -- worgen
 --
 --           NORTHREND
 --
@@ -271,6 +304,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 32413, 4, 72, 80), -- elves
 (NULL, 32413, 7, 72, 80), -- gnomes
 (NULL, 32413, 11, 72, 80), -- draenei
+(NULL, 32413, 12, 72, 80), -- worgen
 --
 -- James Deacon, Valiance Keep, Borean Tundra
 (NULL, 25245, 1, 68, 72), -- human
@@ -278,6 +312,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 25245, 4, 68, 72), -- elves
 (NULL, 25245, 7, 68, 72), -- gnomes
 (NULL, 25245, 11, 68, 72), -- draenei
+(NULL, 25245, 12, 68, 72), -- worgen
 --
 -- "Charlie" Northtop, Fizzcrank Airstrip, Borean Tundra
 (NULL, 26596, 1, 69, 72), -- human
@@ -285,6 +320,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 26596, 4, 69, 72), -- elves
 (NULL, 26596, 7, 69, 72), -- gnomes
 (NULL, 26596, 11, 69, 72), -- draenei
+(NULL, 26596, 12, 69, 72), -- worgen
 --
 -- Innkeeper Hazel Lagras, Valgarde, Howling Fjord
 (NULL, 23731, 1, 68, 72), -- human
@@ -292,6 +328,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 23731, 4, 68, 72), -- elves
 (NULL, 23731, 7, 68, 72), -- gnomes
 (NULL, 23731, 11, 68, 72), -- draenei
+(NULL, 23731, 12, 68, 72), -- worgen
 --
 -- Innkeeper Celeste Goodhutch, Westguard Keep, Howling Fjord
 (NULL, 23937, 1, 69, 72), -- human
@@ -299,6 +336,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 23937, 4, 69, 72), -- elves
 (NULL, 23937, 7, 69, 72), -- gnomes
 (NULL, 23937, 11, 69, 72), -- draenei
+(NULL, 23937, 12, 69, 72), -- draenei
 --
 -- Christina Daniels, Fort Wildervar, Howling Fjord
 (NULL, 24057, 1, 70, 72), -- human
@@ -306,6 +344,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 24057, 4, 70, 72), -- elves
 (NULL, 24057, 7, 70, 72), -- gnomes
 (NULL, 24057, 11, 70, 72), -- draenei
+(NULL, 24057, 12, 70, 72), -- worgen
 --
 -- Jennifer Bell, Amberpine Lodge, Grizzly Hills
 (NULL, 27066, 1, 70, 74), -- human
@@ -313,6 +352,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 27066, 4, 70, 74), -- elves
 (NULL, 27066, 7, 70, 74), -- gnomes
 (NULL, 27066, 11, 70, 74), -- draenei
+(NULL, 27066, 12, 70, 74), -- worgen
 --
 -- Quartermaster McCarty, Westfall Brigade Encampment, Grizzly Hills
 (NULL, 26375, 1, 70, 74), -- human
@@ -320,6 +360,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 26375, 4, 70, 74), -- elves
 (NULL, 26375, 7, 70, 74), -- gnomes
 (NULL, 26375, 11, 70, 74), -- draenei
+(NULL, 26375, 12, 70, 74), -- worgen
 --
 -- Illusia Lune, Wintergarde Keep, Dragonblight
 (NULL, 27042, 1, 71, 75), -- human
@@ -327,6 +368,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 27042, 4, 71, 75), -- elves
 (NULL, 27042, 7, 71, 75), -- gnomes
 (NULL, 27042, 11, 71, 75), -- draenei
+(NULL, 27042, 12, 71, 75), -- worgen
 --
 -- Naohain, Stars' Rest, Dragonblight
 (NULL, 27052, 1, 71, 75), -- human
@@ -334,6 +376,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 27052, 4, 71, 75), -- elves
 (NULL, 27052, 7, 71, 75), -- gnomes
 (NULL, 27052, 11, 71, 75), -- draenei
+(NULL, 27052, 12, 71, 75), -- worgen
 --
 -- Gunda Boldhammer, Frosthold, Storm Peaks
 (NULL, 29926, 1, 77, 80), -- human
@@ -341,6 +384,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 29926, 4, 77, 80), -- elves
 (NULL, 29926, 7, 77, 80), -- gnomes
 (NULL, 29926, 11, 77, 80), -- draenei
+(NULL, 29926, 12, 77, 80), -- worgen
 --
 -- Caris Sunlance, Argent Tournament, Icecrown
 (NULL, 33970, 1, 80, 80), -- human
@@ -348,6 +392,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 33970, 4, 80, 80), -- elves
 (NULL, 33970, 7, 80, 80), -- gnomes
 (NULL, 33970, 11, 80, 80), -- draenei
+(NULL, 33970, 12, 80, 80), -- worgen
 --
 --       ALLIANCE MOUNT VENDORS
 --
@@ -363,11 +408,12 @@ INSERT INTO `playerbots_rpg_races` VALUES
 --
 --       HORDE
 --
---       ORCS & TROLLS
+--       ORCS, TROLLS, & GOBLINS
 --
 -- Innkeeper Grosk, Durotar
 (NULL, 6928, 2, 1, 10),
 (NULL, 6928, 8, 1, 10),
+(NULL, 6928, 9, 1, 10),
 --
 --       TAUREN
 --
@@ -396,6 +442,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 6929, 6, 10, 80), -- tauren
 (NULL, 6929, 5, 20, 80), -- undead
 (NULL, 6929, 10, 20, 80), -- blood elves
+(NULL, 6929, 9, 10, 80), -- goblins
 --
 -- Innkeeper Pala, Thunder Bluff, Mulgore
 (NULL, 6746, 6, 10, 80), -- tauren
@@ -411,6 +458,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 17630, 5, 60, 80), -- undead
 (NULL, 17630, 6, 60, 80), -- tauren
 (NULL, 17630, 8, 60, 80), -- trolls
+(NULL, 17630, 9, 60, 80), -- goblins
 -- Innkeeper Delaniel Silvermoon Entrance
 (NULL, 15433, 10, 5, 7), -- blood elves
 --
@@ -422,23 +470,27 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 3934, 8, 10, 25), -- trolls
 (NULL, 3934, 10, 20, 25), -- blood elves
 -- (NULL, 3934, 5, 15, 25), -- undead
+(NULL, 3934, 9, 10, 25), -- goblins
 --
 -- Innkeeper Byula, Camp Taurajo, Barrens
 (NULL, 7714, 2, 10, 25), -- orcs
 (NULL, 7714, 6, 10, 25), -- tauren
 (NULL, 7714, 8, 10, 25), -- trolls
+(NULL, 7714, 9, 10, 25), -- goblins
 --
 -- Innkeeper Jayka, Stonetalon, Red Rock Retreat
 (NULL, 7731, 2, 15, 27), -- orcs
 (NULL, 7731, 6, 15, 27), -- tauren
 (NULL, 7731, 8, 15, 27), -- trolls
 (NULL, 7731, 10, 20, 27), -- blood elves
+(NULL, 7731, 9, 15, 27), -- goblins
 --
 -- Innkeeper Abeqwa, Thousand Needles
 (NULL, 11116, 2, 25, 35), -- orcs
 (NULL, 11116, 6, 25, 35), -- tauren
 (NULL, 11116, 8, 25, 35), -- trolls
 (NULL, 11116, 10, 25, 35), -- blood elves
+(NULL, 11116, 9, 25, 35), -- goblins
 --
 -- Innkeeper Shay, Tarren Mill, Hillsbrad
 (NULL, 2388, 5, 20, 30), -- undead
@@ -451,15 +503,18 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 12196, 2, 18, 30), -- orcs
 (NULL, 12196, 8, 18, 30), -- trolls
 (NULL, 12196, 10, 20, 30), -- blood elves
+(NULL, 12196, 9, 18, 30), -- goblins
 --
 -- Marukai, Zoram'gar, Ashenvale
 (NULL, 12719, 2, 18, 30), -- orcs
 (NULL, 12719, 8, 18, 30), -- trolls
+(NULL, 12719, 9, 18, 30), -- goblns
 --
 -- Innkeeper Sikewa, Desolace
 (NULL, 11106, 2, 30, 40), -- orcs
 (NULL, 11106, 6, 30, 40), -- tauren
 (NULL, 11106, 8, 30, 40), -- trolls
+(NULL, 11106, 9, 30, 40), -- goblins
 --
 -- Innkeeper Adegwa, Arathi, Hammerfall
 (NULL, 9501, 2, 30, 40), -- orcs
@@ -467,6 +522,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 9501, 6, 30, 40), -- tauren
 (NULL, 9501, 8, 30, 40), -- trolls
 (NULL, 9501, 10, 30, 40), -- blood elves
+(NULL, 9501, 9, 30, 40), -- goblins
 --
 -- Innkeeper Lard, Revantusk Village , Hinterlands
 (NULL, 14731, 2, 40, 50), -- orcs
@@ -474,6 +530,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 14731, 6, 40, 50), -- tauren
 (NULL, 14731, 8, 40, 50), -- trolls
 (NULL, 14731, 10, 40, 50), -- blood elves
+(NULL, 14731, 9, 40, 50), -- goblins
 --
 -- Innkeeper Shul'kar, Kargath Outpost, Badlands
 (NULL, 9356, 2, 35, 45), -- orcs
@@ -481,6 +538,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 9356, 6, 35, 45), -- tauren
 (NULL, 9356, 8, 35, 45), -- trolls
 (NULL, 9356, 10, 35, 45), -- blood elves
+(NULL, 9356, 9, 35, 45), -- goblins
 --
 -- Innkeeper Karakul, Swamp of Sorrows
 (NULL, 6930, 2, 35, 45), -- orcs
@@ -488,6 +546,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 6930, 6, 35, 45), -- tauren
 (NULL, 6930, 8, 35, 45), -- trolls
 (NULL, 6930, 10, 35, 45), -- blood elves
+(NULL, 6930, 9, 35, 45), -- goblins
 --
 -- Innkeeper Thulbek, Grom Gol, Stranglethorn Vale
 (NULL, 5814, 2, 30, 45), -- orcs
@@ -495,17 +554,20 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 5814, 6, 30, 45), -- tauren
 (NULL, 5814, 8, 30, 45), -- trolls
 (NULL, 5814, 10, 30, 45), -- blood elves
+(NULL, 5814, 9, 30, 45), -- goblins
 --
 -- Overlord Mok'Morokk, Dustwallow Marsh
 (NULL, 4500, 2, 35, 45), -- orcs
 (NULL, 4500, 6, 35, 45), -- tauren
 (NULL, 4500, 8, 35, 45), -- trolls
+(NULL, 4500, 9, 35, 45), -- goblins
 --
 -- Jediga, Azshara horde camp
 (NULL, 8587, 2, 45, 55), -- orcs
 (NULL, 8587, 8, 45, 55), -- trolls
 (NULL, 8587, 6, 45, 55), -- tauren
 (NULL, 8587, 10, 45, 55), -- blood elves
+(NULL, 8587, 9, 45, 55), -- goblins
 --
 -- Winna Hazzard, Felwood horde camp
 (NULL, 9996, 2, 48, 55), -- orcs
@@ -513,6 +575,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 9996, 8, 48, 55), -- trolls
 (NULL, 9996, 5, 48, 55), -- undead
 (NULL, 9996, 10, 48, 55), -- blood elves
+(NULL, 9996, 9, 48, 55), -- goblins
 --
 -- Larion, Horde Q-giver, Un'Goro
 (NULL, 9118, 2, 48, 55), -- orcs
@@ -520,6 +583,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 9118, 8, 48, 55), -- trolls
 (NULL, 9118, 5, 48, 55), -- undead
 (NULL, 9118, 10, 48, 55), -- blood elves
+(NULL, 9118, 9, 48, 55), -- goblins
 --
 -- Vahgruk, Horde Taxi, Burning Steppes
 (NULL, 13177, 2, 50, 60), -- orcs
@@ -527,6 +591,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 13177, 6, 50, 60), -- tauren
 (NULL, 13177, 8, 50, 60), -- trolls
 (NULL, 13177, 10, 50, 60), -- blood elves
+(NULL, 13177, 9, 50, 60), -- goblins
 --
 -- General Kirika, Horde camp, Silithus
 (NULL, 17079, 2, 55, 60), -- orcs
@@ -534,6 +599,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 17079, 6, 55, 60), -- tauren
 (NULL, 17079, 8, 55, 60), -- trolls
 (NULL, 17079, 10, 55, 60), -- blood elves
+(NULL, 17079, 9, 55, 60), -- goblins
 --
 --        OUTLAND
 --
@@ -543,6 +609,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 19253, 6, 58, 59), -- tauren
 (NULL, 19253, 8, 58, 59), -- trolls
 (NULL, 19253, 10, 58, 59), -- blood elves
+(NULL, 19253, 9, 58, 59), -- goblins
 --
 -- Floyd Pinkus, Thrallmar, Hellfire
 (NULL, 16602, 2, 58, 63), -- orcs
@@ -550,6 +617,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 16602, 6, 58, 63), -- tauren
 (NULL, 16602, 8, 58, 63), -- trolls
 (NULL, 16602, 10, 58, 63), -- blood elves
+(NULL, 16602, 9, 58, 63), -- goblins
 --
 -- Innkeeper Bazil, Falcon Watch, Hellfire
 (NULL, 18905, 2, 60, 63), -- orcs
@@ -557,6 +625,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 18905, 6, 60, 63), -- tauren
 (NULL, 18905, 8, 60, 63), -- trolls
 (NULL, 18905, 10, 60, 63), -- blood elves
+(NULL, 18905, 9, 60, 63), -- goblins
 --
 -- Innkeeper Merajit, Zabra'jin, Zangarmarsh
 (NULL, 18245, 2, 60, 64), -- orcs
@@ -564,6 +633,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 18245, 6, 60, 64), -- tauren
 (NULL, 18245, 8, 60, 64), -- trolls
 (NULL, 18245, 10, 60, 64), -- blood elves
+(NULL, 18245, 9, 60, 64), -- goblins
 --
 -- Innkeeper Grilka, Terrokar
 (NULL, 18957, 2, 62, 65), -- orcs
@@ -571,6 +641,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 18957, 6, 62, 65), -- tauren
 (NULL, 18957, 8, 62, 65), -- trolls
 (NULL, 18957, 10, 62, 65), -- blood elves
+(NULL, 18957, 9, 62, 65), -- goblins
 --
 -- Matron Tikkit, Garadar, Nagrand
 (NULL, 18913, 2, 62, 65), -- orcs
@@ -578,6 +649,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 18913, 6, 62, 65), -- tauren
 (NULL, 18913, 8, 62, 65), -- trolls
 (NULL, 18913, 10, 62, 65), -- blood elves
+(NULL, 18913, 9, 62, 65), -- goblins
 --
 -- Innkeeper Matron Varah, Mok'Nathal, Blade's Edge
 (NULL, 21088, 2, 65, 68), -- orcs
@@ -585,6 +657,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 21088, 6, 65, 68), -- tauren
 (NULL, 21088, 8, 65, 68), -- trolls
 (NULL, 21088, 10, 65, 68), -- blood elves
+(NULL, 21088, 9, 65, 68), -- goblins
 --
 -- Innkeeper Gholah, Thunderlord, Blade's Edge
 (NULL, 19470, 2, 65, 68), -- orcs
@@ -592,6 +665,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 19470, 6, 65, 68), -- tauren
 (NULL, 19470, 8, 65, 68), -- trolls
 (NULL, 19470, 10, 65, 68), -- blood elves
+(NULL, 19470, 9, 65, 68), -- goblins
 --
 -- Innkeeper Darg Bloodclaw, Shadowmoon Village
 (NULL, 19319, 2, 67, 70), -- orcs
@@ -599,6 +673,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 19319, 6, 67, 70), -- tauren
 (NULL, 19319, 8, 67, 70), -- trolls
 (NULL, 19319, 10, 67, 70), -- blood elves
+(NULL, 19319, 9, 67, 70), -- goblins
 --
 --        NORTHREND
 --
@@ -608,6 +683,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 31557, 6, 72, 80), -- tauren
 (NULL, 31557, 8, 72, 80), -- trolls
 (NULL, 31557, 10, 72, 80), -- blood elves
+(NULL, 31557, 9, 72, 80), -- goblins
 --
 -- Williamson, Warsong Hold, Borean Tundra
 (NULL, 25278, 2, 68, 72), -- orcs
@@ -615,6 +691,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 25278, 6, 68, 72), -- tauren
 (NULL, 25278, 8, 68, 72), -- trolls
 (NULL, 25278, 10, 68, 72), -- blood elves
+(NULL, 25278, 9, 68, 72), -- goblins
 --
 -- Pahu Frosthoof, Taunka'le Village, Borean Tundra
 (NULL, 26709, 2, 69, 72), -- orcs
@@ -622,6 +699,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 26709, 6, 69, 72), -- tauren
 (NULL, 26709, 8, 69, 72), -- trolls
 (NULL, 26709, 10, 69, 72), -- blood elves
+(NULL, 26709, 9, 69, 72), -- goblins
 --
 -- Matron Magah, Bor'Gorok Outpost, Borean Tundra
 (NULL, 26709, 2, 70, 72), -- orcs
@@ -629,6 +707,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 26709, 6, 70, 72), -- tauren
 (NULL, 26709, 8, 70, 72), -- trolls
 (NULL, 26709, 10, 70, 72), -- blood elves
+(NULL, 26709, 9, 70, 72), -- goblins
 --
 -- Basil Osgood, New Agamand, Howling Fjord
 (NULL, 24149, 2, 68, 72), -- orcs
@@ -636,6 +715,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 24149, 6, 68, 72), -- tauren
 (NULL, 24149, 8, 68, 72), -- trolls
 (NULL, 24149, 10, 68, 72), -- blood elves
+(NULL, 24149, 9, 68, 72), -- goblins
 --
 -- Timothy Holland, Vengeance Landing, Howling Fjord
 (NULL, 24342, 2, 69, 72), -- orcs
@@ -643,6 +723,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 24342, 6, 69, 72), -- tauren
 (NULL, 24342, 8, 69, 72), -- trolls
 (NULL, 24342, 10, 69, 72), -- blood elves
+(NULL, 24342, 9, 69, 72), -- goblins
 --
 -- Bori Wintertotem, Camp Winterhoof, Howling Fjord
 (NULL, 24033, 2, 70, 72), -- orcs
@@ -650,6 +731,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 24033, 6, 70, 72), -- tauren
 (NULL, 24033, 8, 70, 72), -- trolls
 (NULL, 24033, 10, 70, 72), -- blood elves
+(NULL, 24033, 9, 70, 72), -- goblins
 --
 -- Barracks Master Rhekku, Conquest Hold, Grizzly Hills
 (NULL, 27125, 2, 70, 74), -- orcs
@@ -657,6 +739,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 27125, 6, 70, 74), -- tauren
 (NULL, 27125, 8, 70, 74), -- trolls
 (NULL, 27125, 10, 70, 74), -- blood elves
+(NULL, 27125, 9, 70, 74), -- goblins
 --
 -- Aiyan Coldwind, Capm Onequah, Grizzly Hills
 (NULL, 26680, 2, 70, 74), -- orcs
@@ -664,6 +747,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 26680, 6, 70, 74), -- tauren
 (NULL, 26680, 8, 70, 74), -- trolls
 (NULL, 26680, 10, 70, 74), -- blood elves
+(NULL, 26680, 9, 70, 74), -- goblins
 --
 -- Mrs. Winterby, Venomspite, Dragonblight
 (NULL, 27027, 2, 71, 75), -- orcs
@@ -671,6 +755,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 27027, 6, 71, 75), -- tauren
 (NULL, 27027, 8, 71, 75), -- trolls
 (NULL, 27027, 10, 71, 75), -- blood elves
+(NULL, 27027, 9, 71, 75), -- goblins
 --
 -- Barracks Master Harga, Agmar's Hammer, Dragonblight
 (NULL, 26985, 2, 71, 75), -- orcs
@@ -678,6 +763,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 26985, 6, 71, 75), -- tauren
 (NULL, 26985, 8, 71, 75), -- trolls
 (NULL, 26985, 10, 71, 75), -- blood elves
+(NULL, 26985, 9, 71, 75), -- goblins
 --
 -- Wabada Whiteflower, Camp Tunka'lo, Storm Peaks
 (NULL, 29971, 2, 77, 80), -- orcs
@@ -685,6 +771,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 29971, 6, 77, 80), -- tauren
 (NULL, 29971, 8, 77, 80), -- trolls
 (NULL, 29971, 10, 77, 80), -- blood elves
+(NULL, 29971, 9, 77, 80), -- goblins
 --
 -- Jarin Dawnglow, Argent Tournament, Icecrown
 (NULL, 33971, 2, 80, 80), -- orcs
@@ -692,6 +779,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 (NULL, 33971, 6, 80, 80), -- tauren
 (NULL, 33971, 8, 80, 80), -- trolls
 (NULL, 33971, 10, 80, 80), -- blood elves
+(NULL, 33971, 9, 80, 80), -- goblins
 --
 --        NEUTRAL AREAS
 --
@@ -700,6 +788,7 @@ INSERT INTO `playerbots_rpg_races` VALUES
 -- Innkeeper Wiley, Ratchet 6791 (Neutral)
 (NULL, 6791, 2, 10, 25),
 (NULL, 6791, 8, 10, 25),
+(NULL, 6791, 9, 10, 25),
 -- Innkeeper Fizzgrimble, Tanaris 7733 (Neutral)
 (NULL, 7733, 0, 40, 50),
 -- Master Smith Burninate, Searing Gorge
diff --git a/src/ChatHelper.cpp b/src/ChatHelper.cpp
index 9ba3337f..83c23075 100644
--- a/src/ChatHelper.cpp
+++ b/src/ChatHelper.cpp
@@ -181,6 +181,8 @@ ChatHelper::ChatHelper(PlayerbotAI* botAI) : PlayerbotAIAware(botAI)
     races[RACE_UNDEAD_PLAYER] = "Undead";
     races[RACE_BLOODELF] = "Blood Elf";
     races[RACE_DRAENEI] = "Draenei";
+	races[RACE_WORGEN] = "Worgen";
+	races[RACE_GOBLIN] = "Goblin";
 }
 
 std::string const ChatHelper::formatMoney(uint32 copper)
diff --git a/src/PlayerbotAI.cpp b/src/PlayerbotAI.cpp
index 5dbe5cb9..0b512ea1 100644
--- a/src/PlayerbotAI.cpp
+++ b/src/PlayerbotAI.cpp
@@ -4119,7 +4119,7 @@ bool PlayerbotAI::canDispel(SpellInfo const* spellInfo, uint32 dispelType)
 bool IsAlliance(uint8 race)
 {
     return race == RACE_HUMAN || race == RACE_DWARF || race == RACE_NIGHTELF || race == RACE_GNOME ||
-           race == RACE_DRAENEI;
+           race == RACE_DRAENEI || race == RACE_WORGEN;
 }
 
 bool PlayerbotAI::HasRealPlayerMaster()
diff --git a/src/RandomPlayerbotFactory.cpp b/src/RandomPlayerbotFactory.cpp
index 291d6825..d6443b51 100644
--- a/src/RandomPlayerbotFactory.cpp
+++ b/src/RandomPlayerbotFactory.cpp
@@ -43,6 +43,10 @@ constexpr RandomPlayerbotFactory::NameRaceAndGender RandomPlayerbotFactory::Comb
             return static_cast<NameRaceAndGender>(static_cast<uint8>(NameRaceAndGender::DraeneiMale) + gender);
         case RACE_BLOODELF:
             return static_cast<NameRaceAndGender>(static_cast<uint8>(NameRaceAndGender::BloodelfMale) + gender);
+		case RACE_WORGEN:
+            return static_cast<NameRaceAndGender>(static_cast<uint8>(NameRaceAndGender::GenericMale) + gender);
+		case RACE_GOBLIN:
+            return static_cast<NameRaceAndGender>(static_cast<uint8>(NameRaceAndGender::OrcMale) + gender);
         default:
             LOG_ERROR("playerbots", "The race with ID %d does not have a naming category", race);
             return static_cast<NameRaceAndGender>(static_cast<uint8>(NameRaceAndGender::GenericMale) + gender);
@@ -65,6 +69,8 @@ RandomPlayerbotFactory::RandomPlayerbotFactory(uint32 accountId) : accountId(acc
     {
         availableRaces[CLASS_WARRIOR].push_back(RACE_DRAENEI);
     }
+	availableRaces[CLASS_WARRIOR].push_back(RACE_WORGEN);
+	availableRaces[CLASS_WARRIOR].push_back(RACE_GOBLIN);
 
     availableRaces[CLASS_PALADIN].push_back(RACE_HUMAN);
     availableRaces[CLASS_PALADIN].push_back(RACE_DWARF);
@@ -85,6 +91,8 @@ RandomPlayerbotFactory::RandomPlayerbotFactory(uint32 accountId) : accountId(acc
     {
         availableRaces[CLASS_ROGUE].push_back(RACE_BLOODELF);
     }
+	availableRaces[CLASS_ROGUE].push_back(RACE_WORGEN);
+	availableRaces[CLASS_ROGUE].push_back(RACE_GOBLIN);
 
     availableRaces[CLASS_PRIEST].push_back(RACE_HUMAN);
     availableRaces[CLASS_PRIEST].push_back(RACE_DWARF);
@@ -96,6 +104,8 @@ RandomPlayerbotFactory::RandomPlayerbotFactory(uint32 accountId) : accountId(acc
         availableRaces[CLASS_PRIEST].push_back(RACE_DRAENEI);
         availableRaces[CLASS_PRIEST].push_back(RACE_BLOODELF);
     }
+	availableRaces[CLASS_PRIEST].push_back(RACE_WORGEN);
+	availableRaces[CLASS_PRIEST].push_back(RACE_GOBLIN);
 
     availableRaces[CLASS_MAGE].push_back(RACE_HUMAN);
     availableRaces[CLASS_MAGE].push_back(RACE_GNOME);
@@ -106,6 +116,8 @@ RandomPlayerbotFactory::RandomPlayerbotFactory(uint32 accountId) : accountId(acc
         availableRaces[CLASS_MAGE].push_back(RACE_DRAENEI);
         availableRaces[CLASS_MAGE].push_back(RACE_BLOODELF);
     }
+	availableRaces[CLASS_MAGE].push_back(RACE_WORGEN);
+	availableRaces[CLASS_MAGE].push_back(RACE_GOBLIN);
 
     availableRaces[CLASS_WARLOCK].push_back(RACE_HUMAN);
     availableRaces[CLASS_WARLOCK].push_back(RACE_GNOME);
@@ -115,6 +127,8 @@ RandomPlayerbotFactory::RandomPlayerbotFactory(uint32 accountId) : accountId(acc
     {
         availableRaces[CLASS_WARLOCK].push_back(RACE_BLOODELF);
     }
+	availableRaces[CLASS_WARLOCK].push_back(RACE_WORGEN);
+	availableRaces[CLASS_WARLOCK].push_back(RACE_GOBLIN);
 
     availableRaces[CLASS_SHAMAN].push_back(RACE_ORC);
     availableRaces[CLASS_SHAMAN].push_back(RACE_TAUREN);
@@ -123,6 +137,7 @@ RandomPlayerbotFactory::RandomPlayerbotFactory(uint32 accountId) : accountId(acc
     {
         availableRaces[CLASS_SHAMAN].push_back(RACE_DRAENEI);
     }
+	availableRaces[CLASS_SHAMAN].push_back(RACE_GOBLIN);
 
     availableRaces[CLASS_HUNTER].push_back(RACE_DWARF);
     availableRaces[CLASS_HUNTER].push_back(RACE_NIGHTELF);
@@ -134,9 +149,12 @@ RandomPlayerbotFactory::RandomPlayerbotFactory(uint32 accountId) : accountId(acc
         availableRaces[CLASS_HUNTER].push_back(RACE_DRAENEI);
         availableRaces[CLASS_HUNTER].push_back(RACE_BLOODELF);
     }
+	availableRaces[CLASS_HUNTER].push_back(RACE_WORGEN);
+	availableRaces[CLASS_HUNTER].push_back(RACE_GOBLIN);
 
     availableRaces[CLASS_DRUID].push_back(RACE_NIGHTELF);
     availableRaces[CLASS_DRUID].push_back(RACE_TAUREN);
+	availableRaces[CLASS_DRUID].push_back(RACE_WORGEN);
 
     if (expansion == EXPANSION_WRATH_OF_THE_LICH_KING)
     {
@@ -150,6 +168,8 @@ RandomPlayerbotFactory::RandomPlayerbotFactory(uint32 accountId) : accountId(acc
         availableRaces[CLASS_DEATH_KNIGHT].push_back(RACE_DRAENEI);
         availableRaces[CLASS_DEATH_KNIGHT].push_back(RACE_GNOME);
         availableRaces[CLASS_DEATH_KNIGHT].push_back(RACE_DWARF);
+		availableRaces[CLASS_DEATH_KNIGHT].push_back(RACE_WORGEN);
+		availableRaces[CLASS_DEATH_KNIGHT].push_back(RACE_GOBLIN);
     }
 }
 
diff --git a/src/RandomPlayerbotMgr.cpp b/src/RandomPlayerbotMgr.cpp
index 1b69d7f3..cc3b9a27 100644
--- a/src/RandomPlayerbotMgr.cpp
+++ b/src/RandomPlayerbotMgr.cpp
@@ -2099,7 +2099,7 @@ void RandomPlayerbotMgr::PrepareAddclassCache()
                     Field* fields = results->Fetch();
                     ObjectGuid guid = ObjectGuid(HighGuid::Player, fields[0].Get<uint32>());
                     uint32 race = fields[1].Get<uint32>();
-                    bool isAlliance = race == 1 || race == 3 || race == 4 || race == 7 || race == 11;
+                    bool isAlliance = race == 1 || race == 3 || race == 4 || race == 7 || race == 11 || race == 12;
                     addclassCache[GetTeamClassIdx(isAlliance, claz)].insert(guid);
                     collected++;
                 } while (results->NextRow());
diff --git a/src/TravelMgr.cpp b/src/TravelMgr.cpp
index 69d510b9..15f2c9ae 100644
--- a/src/TravelMgr.cpp
+++ b/src/TravelMgr.cpp
@@ -2459,6 +2459,8 @@ void TravelMgr::LoadQuestTravelTable()
         startNames[RACE_TROLL] = "Orc and Troll";
         startNames[RACE_DRAENEI] = "Draenei";
         startNames[RACE_BLOODELF] = "Blood Elf";
+		startNames[RACE_WORGEN] = "Worgen";
+		startNames[RACE_GOBLIN] = "Goblin";
 
         for (uint32 i = 0; i < MAX_RACES; i++)
         {
diff --git a/src/TravelNode.cpp b/src/TravelNode.cpp
index 92f8d152..bfb96af8 100644
--- a/src/TravelNode.cpp
+++ b/src/TravelNode.cpp
@@ -1685,6 +1685,8 @@ void TravelNodeMap::generateStartNodes()
     startNames[RACE_TAUREN] = "Tauren";
     startNames[RACE_GNOME] = "Dwarf and Gnome";
     startNames[RACE_TROLL] = "Orc and Troll";
+	startNames[RACE_WORGEN] = "Night Elf";
+	startNames[RACE_GOBLIN] = "Orc and Troll";
 
     for (uint32 i = 0; i < MAX_RACES; i++)
     {
diff --git a/src/factory/PlayerbotFactory.cpp b/src/factory/PlayerbotFactory.cpp
index e239d449..3a7dee17 100644
--- a/src/factory/PlayerbotFactory.cpp
+++ b/src/factory/PlayerbotFactory.cpp
@@ -3088,6 +3088,14 @@ void PlayerbotFactory::InitMounts()
             slow = {33660, 35020, 35022, 35018};
             fast = {35025, 35025, 35027};
             break;
+		case RACE_WORGEN:
+            slow = {103195};
+            fast = {103196};
+            break;
+		case RACE_GOBLIN:
+            slow = {87090};
+            fast = {87091};
+            break;
     }
 
     switch (bot->GetTeamId())
diff --git a/src/strategy/actions/EmoteAction.cpp b/src/strategy/actions/EmoteAction.cpp
index ba44424b..92da9b4b 100644
--- a/src/strategy/actions/EmoteAction.cpp
+++ b/src/strategy/actions/EmoteAction.cpp
@@ -892,6 +892,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                 return 2;
             }
             case RACE_GNOME:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 1;
@@ -949,6 +951,20 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                     return 3;
 
                 return 3;
+            }
+			case RACE_WORGEN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 5;
+
+                return 4;
+            }
+			case RACE_GOBLIN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 5;
+
+                return 5;
             }
         }
     }
@@ -976,6 +992,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                 return 2;
             }
             case RACE_GNOME:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 3;
@@ -1013,6 +1031,14 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                     return 3;
 
                 return 3;
+            }
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 4;
+
+                return 4;
             }
         }
     }
@@ -1049,6 +1075,14 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                     return 2;
 
                 return 1;
+            }
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 3;
+
+                return 3;
             }
         }
     }
@@ -1076,6 +1110,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
             case RACE_GNOME:
             case RACE_ORC:
             case RACE_UNDEAD_PLAYER:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 2;
@@ -1094,6 +1130,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
             case RACE_ORC:
             case RACE_UNDEAD_PLAYER:
             case RACE_TAUREN:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 2;
@@ -1120,6 +1158,7 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
             case RACE_ORC:
             case RACE_UNDEAD_PLAYER:
             case RACE_TAUREN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 2;
@@ -1133,6 +1172,13 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                     return 3;
 
                 return 2;
+            }
+			case RACE_WORGEN:
+			{
+                if (Gender == GENDER_MALE)
+                    return 2;
+
+                return 3;
             }
         }
     }
@@ -1147,6 +1193,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
             case RACE_TAUREN:
             case RACE_TROLL:
             case RACE_GNOME:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 return 2;
             }
@@ -1177,6 +1225,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                 return 3;
             }
             case RACE_GNOME:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 4;
@@ -1214,6 +1264,20 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                     return 4;
 
                 return 4;
+            }
+			case RACE_WORGEN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 6;
+
+                return 5;
+            }
+			case RACE_GOBLIN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 6;
+
+                return 6;
             }
         }
     }
@@ -1235,6 +1299,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
             case RACE_ORC:
             case RACE_UNDEAD_PLAYER:
             case RACE_TAUREN:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 3;
@@ -1253,6 +1319,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
             case RACE_ORC:
             case RACE_UNDEAD_PLAYER:
             case RACE_TROLL:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 3;
@@ -1286,6 +1354,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
             case RACE_ORC:
             case RACE_TAUREN:
             case RACE_TROLL:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 3;
@@ -1308,6 +1378,8 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
             case RACE_ORC:
             case RACE_TAUREN:
             case RACE_TROLL:
+			case RACE_WORGEN:
+			case RACE_GOBLIN:
             {
                 if (Gender == GENDER_MALE)
                     return 3;
@@ -1371,6 +1443,20 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                     return 6;
 
                 return 6;
+            }
+			case RACE_WORGEN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 10;
+
+                return 10;
+            }
+			case RACE_GOBLIN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 10;
+
+                return 10;
             }
         }
     }
@@ -1433,6 +1519,20 @@ uint32 EmoteActionBase::GetNumberOfEmoteVariants(TextEmotes emote, uint8 Race, u
                     return 4;
 
                 return 7;
+            }
+			case RACE_WORGEN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 10;
+
+                return 9;
+            }
+			case RACE_GOBLIN:
+            {
+                if (Gender == GENDER_MALE)
+                    return 10;
+
+                return 10;
             }
         }
     }
diff --git a/src/strategy/actions/ReviveFromCorpseAction.cpp b/src/strategy/actions/ReviveFromCorpseAction.cpp
index 960f438f..ff647a0b 100644
--- a/src/strategy/actions/ReviveFromCorpseAction.cpp
+++ b/src/strategy/actions/ReviveFromCorpseAction.cpp
@@ -251,9 +251,9 @@ GraveyardStruct const* SpiritHealerAction::GetGrave(bool startZone)
     std::vector<uint32> races;
 
     if (bot->GetTeamId() == TEAM_ALLIANCE)
-        races = {RACE_HUMAN, RACE_DWARF, RACE_GNOME, RACE_NIGHTELF};
+        races = {RACE_HUMAN, RACE_DWARF, RACE_GNOME, RACE_NIGHTELF, RACE_WORGEN};
     else
-        races = {RACE_ORC, RACE_TROLL, RACE_TAUREN, RACE_UNDEAD_PLAYER};
+        races = {RACE_ORC, RACE_TROLL, RACE_TAUREN, RACE_UNDEAD_PLAYER, RACE_GOBLIN};
 
     float graveDistance = -1;
 
