diff --git a/src/PlayerbotAI.cpp b/src/PlayerbotAI.cpp
index 2b2bc89b..76344de4 100644
--- a/src/PlayerbotAI.cpp
+++ b/src/PlayerbotAI.cpp
@@ -4119,7 +4119,7 @@ bool PlayerbotAI::canDispel(SpellInfo const* spellInfo, uint32 dispelType)
 bool IsAlliance(uint8 race)
 {
     return race == RACE_HUMAN || race == RACE_DWARF || race == RACE_NIGHTELF || race == RACE_GNOME ||
-           race == RACE_DRAENEI;
+           race == RACE_DRAENEI || race == RACE_WORGEN;
 }
 
 bool PlayerbotAI::HasRealPlayerMaster()
